### PR TITLE
HOTFIX: Handle primitive types in AvroFormat

### PR DESCRIPTION
**Pull request by kkonstantine**
* Follows from: https://github.com/confluentinc/kafka-connect-hdfs/issues/130
 * Equivalent to: https://github.com/confluentinc/kafka-connect-hdfs/pull/176

### CODE REVIEW

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java [hunk 1]
```diff
@@ -74,7 +75,13 @@ public void write(SinkRecord record) {
...
         Object value = avroData.fromConnectData(schema, record.value());
         try {
-          writer.append(value);
+          // AvroData wraps primitive types so their schema can be included. We need to unwrap
+          // NonRecordContainers to just their value to properly handle these types
+          if (value instanceof NonRecordContainer) {
+            writer.append(((NonRecordContainer) value).getValue());
+          } else {
+            writer.append(value);
+          }
```
Comment (rhauch) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java]: Would it make more sense to do something like this inside the try block? (I've removed comments to keep things simple.)

```
if (value instanceof NonRecordContainer) {
    // put comment here
    value = ((NonRecordContainer) value).getValue();
}
writer.append(value);
```

Comment (rhauch) [reply kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java]: Just to clarify, this wrapping in a `NonRecordContainer` only happens at the top-level value in a record?

Comment (rhauch) [reply kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java]: BTW, I do see that this is the same code that was added in https://github.com/confluentinc/kafka-connect-hdfs/pull/176, but still seems like that is a bit more complicated.

Comment (kkonstantine) [reply kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java]: Actually, that's how I wrote it in the test I'm writing now. I like that more too.

Comment (kkonstantine) [reply kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java]: And also, yes, my understanding is that there's no nested application of `NonRecordContainer`.

**Pull request merged**